Cache/Requests.lean

```diff
@@ -0,0 +1,102 @@
+import Cache.Hashing
+
+namespace Cache.Requests
+
+/-- Azure blob URL -/
+def URL : String :=
+  "https://lakecache.blob.core.windows.net/mathlib4"
+
+open System
+
+/-- Retrieves the azure token from the file system -/
+def getToken : IO String :=
+  return (← IO.FS.readFile ⟨"azure.token"⟩).trim
+
+/-- Gets the set of file names hosted on the the server -/
+def getHostedCacheSet : IO $ Std.RBSet String compare := do
+  IO.println "Downloading list hosted files"
+  let ret ← IO.runCmd s!"curl -X GET {URL}?comp=list&restype=container"
```

Comment:
This is going to be a *huge* list after a while.  I think it would be better to just do a `HEAD` request for each file before uploading.  (Or is there a flag to `PUT` that fails if the file already exists?)

Reply:
I suspect this API is designed to handle huge lists more efficiently than multiple smaller. The `PUT` that fails is problematic when we want to manually fix corrupted files

Reply:
Here's a plan: if we notice that `put` is choking when downloading that list, we can rename `put!` to `put` and get rid of this current `put`

Reply:
The mathlib3 olean cache now has more than 20k files, and that's with only one file per revision.  With more fine-grained caching we'll easily get 10x (or more).  So the question here is "download a list of 200k files" vs. "make 3k HEAD requests".

> The PUT that fails is problematic when we want to manually fix corrupted files

I think the flag is an HTTP header called `If-None-Match: *`.  And we can always leave it out to force the update.